dashboard/app/models/pd/enrollment.rb

```diff
@@ -183,6 +192,16 @@ def full_name=(value)
 
   protected
 
+  def enroll_in_corresponding_online_learning
+    plc_course = Plc::Course.find_by(
+      name: WORKSHOP_COURSE_ONLINE_LEARNING_MAPPING[workshop.course]
+    )
```

Comment:
Like above, I think this find logic better belongs in the Workshop model. Encapsulate this in a method, something like `workshop.associated_online_course` that returns the matching PLC course or nil.